pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -609,6 +615,35 @@ func (r *Store) getDynamicField(data map[string]interface{}) (string, error) {
 	return "", nil
 }
 
+func (r *Store) updateClusterByK8sclient(id, name string, data map[string]interface{}) (map[string]interface{}, error) {
+	if r.ClusterClient == nil {
+		return nil, fmt.Errorf("Updating cluster [%s] using K8s dynamic client: k8s client is nil", id)
+	}
+
+	logrus.Tracef("Updating cluster [%s] using K8s dynamic client", id)
+	object, err := r.ClusterClient.Get(context.TODO(), id, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("Getting cluster [%s] using K8s dynamic client: %v", id, err)
+	}
+
+	// Replacing name by displayName to properly update cluster using k8s format
+	values.PutValue(data, name, "displayName")
+	values.RemoveValue(data, "name")
+
+	// Setting data as cluster spec before update
+	object.Object["spec"] = data
+	object, err = r.ClusterClient.Update(context.TODO(), object, metav1.UpdateOptions{})
+	if err != nil || object == nil {
+		return nil, fmt.Errorf("Updating cluster [%s] using K8s dynamic client: %v", id, err)
```

Comment:
Can we change to `Error updating cluster ...` so that we indicate there was an error

Reply:
Updated